KFSPTS-5252 - get latest financials base code and update overlay pom/…

bhutchinson opened the pull request:
…code so it will compile

bhutchinson commented:
@kyleoliveira would probably be good for you to take a look at this too; once it gets merged, you can base the new environment builds off the post6-upgrade branch.

jhulslander commented on src/main/java/edu/cornell/kfs/coa/document/CuSubAccountMaintainableImpl.java:
```diff
@@ -358,10 +358,10 @@ private boolean isIcrSectionDataChanged(List<A21IndirectCostRecoveryAccount> new
      */
     @Override
-    public void processAfterCopy(org.kuali.rice.kns.document.MaintenanceDocument  document, Map<String, String[]> parameters) {
+    public void processAfterCopy(MaintenanceDocument  document, Map<String, String[]> parameters) {
```
The old import was                    org.kuali.rice.krad.maintenance.MaintenanceDocument
the old full qualified object was org.kuali.rice.kns.document.MaintenanceDocument
The new import is                     org.kuali.kfs.kns.document.MaintenanceDocument

Do we need this to be fully qualified to org.kuali.kfs.krad.document.MaintenanceDocument (if it exists)

bhutchinson replied:
org.kuali.kfs.kns.document.MaintenanceDocument extends org.kuali.kfs.krad.document.MaintenanceDocument, so I think this is ok, at least for now; if we find we need to change something to get the application to start/work, we can do it then I think. Thanks.

jhulslander replied:
Sounds reasonable.

jhulslander replied:
I noted this difference on a related Jira, so this difference doesn't get lost.

jhulslander commented on src/main/java/edu/cornell/kfs/coa/document/CuSubAccountMaintainableImpl.java:
```diff
@@ -375,33 +375,33 @@ public void processAfterCopy(org.kuali.rice.kns.document.MaintenanceDocument  do
     }
     
-        public void refresh(String refreshCaller, Map fieldValues, org.kuali.rice.kns.document.MaintenanceDocument document) {
+    public void refresh(String refreshCaller, Map fieldValues, MaintenanceDocument document) {
```
Same concern about krad versus kns

jhulslander commented on src/main/java/edu/cornell/kfs/coa/document/CuSubAccountMaintainableImpl.java:
```diff
@@ -375,33 +375,33 @@ public void processAfterCopy(org.kuali.rice.kns.document.MaintenanceDocument  do
     @SuppressWarnings("rawtypes")
     @Override
-    public List getSections(org.kuali.rice.kns.document.MaintenanceDocument document, Maintainable oldMaintainable) {
+    public List getSections(MaintenanceDocument document, Maintainable oldMaintainable) {
```
Same concern about krad versus kns

jhulslander commented on src/main/java/edu/cornell/kfs/fp/document/authorization/CuProcurementCardAccountingLineAuthorizer.java:
```diff
@@ -39,7 +39,7 @@ protected boolean determineEditPermissionByFieldName(AccountingDocument accounti
         if (accountingDocument.getDocumentHeader() != null &&
                 accountingDocument.getDocumentHeader().getWorkflowDocument() != null) {
-            Set<String> currentNodes = accountingDocument.getDocumentHeader().getWorkflowDocument().getCurrentNodeNames();
+//            Set<String> currentNodes = accountingDocument.getDocumentHeader().getWorkflowDocument().getCurrentNodeNames();
```
since this is commented out, can we just remove the line?

bhutchinson replied:
I suppose. :-) will do, thanks.

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuPurchaseOrderServiceImpl.java:
```diff
@@ -21,25 +21,24 @@
 public class CuPurchaseOrderServiceImpl extends PurchaseOrderServiceImpl {
     private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuPurchaseOrderServiceImpl.class);
 
-    @Override
```
autoCloseFullyDisencumberedOrders no longer exists in the parent?

bhutchinson replied:
good catch, looks like it moved to a new service class - I'll fix, thanks

bhutchinson commented:
I have addressed the comments, thanks. And FWIW, the unit test PR suite fails as I expect it to, there will be more work to do before those will pass.

jhulslander commented:
Issues I noted have been addressed and look good to me.  I will merge after @kyleoliveira takes look too.

kyleoliveira commented:
Looks good to me.
